fecgov/openFEC
# Add filter by case citations in /legal/search/

## Summary
In order to create the legal web search app, we need add filter by citations(statutory citation and (regulatory_citation) in MUR's search. This PR will add general filter by **case_citations** which used for 3 case types: murs, admin_fines and adrs(not release yet)
- Endpoint url: /legal/search/?type=murs&case_statutory_citation=xxxxx&case_regulatory_citation=yyyyy
- Three Parameters: 
case_statutory_citation, 
case_regulatory_citation, 
case_citation_require_all
<img width="705" alt="mur_citation" src="https://user-images.githubusercontent.com/24395751/171548119-365cb808-64b9-42be-ac32-98d28cac2eb8.png">


- Resolves #5111 
- Ref design ticket [https://github.com/fecgov/fec-epics/issues/221](https://github.com/fecgov/fec-epics/issues/221)
- Ref research ticket [https://github.com/fecgov/openFEC/issues/5107](https://github.com/fecgov/openFEC/issues/5107)
- All parameters information: [https://docs.google.com/spreadsheets/d/1l3YKu-oBzOwBHH986YdyiYaSJTFdJBNxitZkt_Ew4zA/edit#gid=1161233667](https://docs.google.com/spreadsheets/d/1l3YKu-oBzOwBHH986YdyiYaSJTFdJBNxitZkt_Ew4zA/edit#gid=1161233667)
- Ref AO search citations page: [https://www.fec.gov/data/legal/search/advisory-opinions/?type=advisory_opinions&ao_category=F&ao_statutory_citation=3352+U.S.C.+%C2%A730104](https://www.fec.gov/data/legal/search/advisory-opinions/?type=advisory_opinions&ao_category=F&ao_statutory_citation=3352+U.S.C.+%C2%A730104)

### Completion criteria
- [x] Create filter by case_citations
- [ ] Recreate ES index("docs") and reload whole data
- [ ] Create follow-up ticket to reload ES data on stage and prod spaces
- [ ] Sync with team 

### Required reviewers
2 developers

## Impacted areas of the application
legal search

## How to test
(It is better test locally first)
- Follow wiki install ES7.4 locally and start it. [https://github.com/fecgov/openFEC/wiki/Elasticsearch-7.x.0-setup-and-test-locally](https://github.com/fecgov/openFEC/wiki/Elasticsearch-7.x.0-setup-and-test-locally)
- Check branch
- Create index "docs"
`python manage.py create_index`
- Upload 8 sample murs in local ES
```
python manage.py load_current_murs -s 7958
python manage.py load_current_murs -s 7949
python manage.py load_current_murs -s 7927
python manage.py load_current_murs -s 7926
python manage.py load_current_murs -s 7923
python manage.py load_current_murs -s 7212
python manage.py load_current_murs -s 4633 
python manage.py load_current_murs -s 3987
```
- Make sure /legal/search/ endpoint works well
- Test filter
Ref test urls list: [https://docs.google.com/spreadsheets/d/1l3YKu-oBzOwBHH986YdyiYaSJTFdJBNxitZkt_Ew4zA/edit#gid=1143741195](https://docs.google.com/spreadsheets/d/1l3YKu-oBzOwBHH986YdyiYaSJTFdJBNxitZkt_Ew4zA/edit#gid=1143741195)
- Some sample urls
`http://127.0.0.1:5000/v1/legal/search/?type=murs&case_statutory_citation=52 U.S.C. §30116&case_statutory_citation=52 U.S.C. §30122&case_regulatory_citation=11 CFR §110.4`

`http://127.0.0.1:5000/v1/legal/search/?type=murs&case_statutory_citation=52 U.S.C. §30116&case_statutory_citation=52 U.S.C. §30122&case_regulatory_citation=11 CFR §110.4&case_citation_require_all=True`

`http://127.0.0.1:5000/v1/legal/search/?type=murs&case_statutory_citation=52 U.S.C. §30104&case_regulatory_citation=11 CFR §116.10&case_citation_require_all=True`
